src/Widgets/PowerModeList.vala

```diff
@@ -0,0 +1,170 @@
+/*
+ * SPDX-License-Identifier: GPL-2.0-or-later
+ * SPDX-FileCopyrightText: 2024 elementary, Inc. (https://elementary.io)
+ */
+
+using Power.Services.DBusInterfaces;
+
+public class Power.Widgets.PowerModeList : Gtk.Box {
+    public static bool successfully_initialized { get; private set; default = true; }
+
+    private static PowerProfile? pprofile;
+    private static GLib.Settings? settings;
+
+    public string settings_key { get; construct; }
+
+    private Gtk.RadioButton saver_radio;
+    private Gtk.RadioButton balanced_radio;
+    private Gtk.RadioButton performance_radio;
+
+    private const string ICON_RES = "/io/elementary/desktop/wingpanel/power/scalable/categories/";
+
+    public PowerModeList (bool on_battery) {
+        Object (
+            settings_key: on_battery ? "profile-on-good-battery" : "profile-plugged-in",
+            orientation: Gtk.Orientation.VERTICAL,
+            spacing: 0,
+            margin: 6
+        );
+    }
+
+    static construct {
+        var schema = SettingsSchemaSource.get_default ().lookup ("io.elementary.settings-daemon.power", true);
+        if (schema != null && schema.has_key ("profile-plugged-in") && schema.has_key ("profile-on-good-battery")) {
+            settings = new GLib.Settings ("io.elementary.settings-daemon.power");
+        } else {
+            warning ("settings-daemon schema not found, will connect to power-profiles-daemon directly");
+        }
+
+        try {
+            pprofile = Bus.get_proxy_sync (BusType.SYSTEM, POWER_PROFILES_DAEMON_NAME, POWER_PROFILES_DAEMON_PATH, DBusProxyFlags.NONE);
+        } catch (Error e) {
+            critical (e.message);
+            successfully_initialized = false;
+        }
+    }
+
+    construct {
+        if (pprofile == null) {
+            return;
+        }
+
+        build_ui ();
+        build_events ();
+    }
+
+    private void build_ui () {
+        var saver_icon = new Gtk.Image.from_resource (ICON_RES + "power-mode-powersaver-symbolic.svg");
```

Comment:
Instead of adding icons from resource like this, you can just add them to the icon theme and then reference them by name. So in the Indicator class you could add:
`
Gtk.IconTheme.get_default ().add_resource_path ("/io/elementary/desktop/wingpanel/power");
`

Reply:
Fixed